dragonfly_doe2/doe/floor_space.py

```diff
@@ -0,0 +1,149 @@
+from dragonfly.room2d import Room2D
+from dragonfly.story import Story
+from dataclasses import dataclass
+from typing import List
+
+
+@dataclass
+class Wall:
+    """*.inp Wall object.
+    Example:
+        .. code-block::f#
+            "simple_example_dfb_Floor1_Room1 Wall_1" = EXTERIOR-WALL
+               CONSTRUCTION    = "EWall Construction"
+               LOCATION        = SPACE-V1
+               ..
+    """
+    name: str = None
+    location_index: int = None
+    wall_con_name: str = None
+
+    @classmethod
+    def from_room_seg(cls, _name: str, _bc_id: int, _wall_con_name: str):
+        indexed_id = _bc_id+1
+        return cls(str(_name+f'_Wall_{indexed_id}'), indexed_id, _wall_con_name)
```

Comment:
A formated string is already a string - also if you are using a formatted string then you should put all the variables there:

```suggestion
        return cls(f'{_name}_Wall_{indexed_id}', indexed_id, _wall_con_name)
```